### PR TITLE
Toolbox Mastery: Finally gives toolboxes a real use

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Using a toolbox on an object will now bring up a radial menu containing all "tools" inside of it, and picking one will put it in your offhand and use it on the object you clicked. After use, the item will be stowed back in the toolbox. Welding tools will be turned on (and off when stowed), and spess knives will give you an option to pick which tool you want to use. Only certain items qualify as tools (most items that would fit into a toolbelt do), so you cannot use this to sneakily attack someone with an esword.

https://github.com/user-attachments/assets/0cd5c859-490f-4643-b451-92ce15a34fba

Using a toolbox in such a fashion passes click modifiers through, allowing you to use left/right, ctrl and alt click modes of items just fine.

God bless our attack chain.

## Why It's Good For The Game

Toolboxes are inferior to toolbelts and boxes in almost every single way, being only capable of storing 7 items and not fitting anywhere. This gives them an actual use and maybe they'll see use in actual construction projects.

## Changelog
:cl:
add: Toolboxes can be used on any object to pull out and use a tool from it as long as your offhand is free.
/:cl:

### CODE REVIEW

**Comment by DrAmazing343** [comment]
This is like seeing the fucking Renaissance happening in front of my eyes bros this is HUGE. Like, can we step back and think for a second about how for fucking ages, the game's been seen as the pinnacle of granular "click 9 things to make one thing happen" and now toolboxes are going to have goddamn RADIAL MENUS??? This is some ERGONOMICS and I'm all for it to make toolboxes more properly useful.

**Comment by Dawnseer** [comment]
It's actually amazing, this is gonna be huge

**Comment by SmArtKar** [comment]
> it might be a big ask, but it would be awesome QOL if you could also use the same tool on multiple tiles adjacent to you (e.g., right clicking uses the last tool you selected in the toolbox if that tool doesn't already have a right click handler)

Doesn't even have to be right click actually, since you're still holding the tool

**Comment by SmArtKar** [comment]
https://github.com/user-attachments/assets/cea4feb7-cfdf-4420-8416-42df4e6d5ac1

File: code/game/objects/items/storage/toolbox.dm [hunk 1]
```diff
@@ -37,6 +39,71 @@
...
+	if (user.get_inactive_held_item())
+		user.balloon_alert(user, "hands busy!")
+		return ITEM_INTERACT_BLOCKING
+
+	var/list/item_radial = list()
+	for (var/obj/item/tool in atom_storage.real_location)
+		for (var/item_type in GLOB.tool_items)
+			if (istype(tool, item_type))
+				item_radial[tool] = tool.appearance
+				break
```
Comment (MrMelbert) [inline code/game/objects/items/storage/toolbox.dm:62]:
```suggestion
	for (var/obj/item/tool in atom_storage.real_location)
		if(is_type_in_list(tool, GLOB.tool_items))
			item_radial[tool] = tool.appearance
			break
```

**Comment by SmArtKar** [comment]
> maybe add toolbelts somethings like that but with action button?

The whole point is to give toolboxes an advantage over toolbelts, that would just make them redundant again.

**Comment by GuillaumePrata** [comment]
I'm never using a toolbelt again once this is merged.

**Comment by Namwty** [comment]
This is great, but how will it work, if ya let's say got blown up by fuel tank and lost a limb ( offhand ). I assume just like regular storage item, via alt click? Will it just drop the tool on the floor, if ya use toolbox this way?

**Comment by MrMelbert** [comment]
im confused, toolboxes have a use, they kill people

**Comment by Loafin34** [comment]
welding tool does more damage, fire extinguisher does the same damage and is normal, and screwdriver is the king. toolboxes do jack sheeeet

**Comment by MrMelbert** [comment]
fire exes certainly don't have the same force as a toolbox

also toolboxes cause brain damage which gives you the moral high groud

**Comment by Loafin34** [comment]
true, but they're also bulky and everything else i said is normal size. as for maintenance weaponry, it is defeated by the lead pipe and arguably potted plant throwing. If you test this in a deathmatch, you'll note that toolboxes themselves are rarely used, instead vying for the tools within.

**Comment by SmArtKar** [comment]
> This is great, but how will it work, if ya let's say got blown up by fuel tank and lost a limb ( offhand ). I assume just like regular storage item, via alt click? Will it just drop the tool on the floor, if ya use toolbox this way?

You cannot pull a tool out with your second hand if you don't have one. (Nothing will happen)

**Comment by carlarctg** [comment]
real talk how do we fit crowbars inside a toolbox how does that make any sense

**Comment by carlarctg** [comment]
erm thry shoiñf ne remamed....


give this functionality to surgery trays

**Comment by SmArtKar** [comment]
> erm thry shoiñf ne remamed....
> 
> give this functionality to surgery trays

Those will receive separate functionality soon:tm:

**Pull request merged**